iot-hub/Tutorials/Routing/SimulatedDevice.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+// Pass in the device connection string to authenticate the device with your IoT hub.
+// Using the Azure CLI:
+// az iot hub device-identity show-connection-string --hub-name {YourIoTHubName} --device-id MyTestDevice --output table
+var connectionString = process.argv[2];
+
+// Using the Node.js Device SDK for IoT Hub:
+//   https://github.com/Azure/azure-iot-sdk-node
+// The sample connects to a device-specific MQTT endpoint on your IoT Hub.
+var Protocol = require('azure-iot-device-mqtt').Mqtt;
+var Client = require('azure-iot-device').Client;
+var Message = require('azure-iot-device').Message;
+var NoRetry = require('azure-iot-common').NoRetry;
+var client = Client.fromConnectionString(connectionString, Protocol);
+const chalk = require('chalk');
+
+console.log('IoT Hub troubleshooting tutorial\nSimulated device #3\n')
+
+// Print results.
+function printResultFor(op) {
+  return function printResult(err, res) {
+    if (err) console.log(op + ' error: ' + err.toString());
+    if (res) console.log(op + ' status: ' + res.constructor.name);
+  };
+}
+
+// Connect to the IoT hub.
+client.open(function (err) {
+  if (err) {
+    console.log(chalk.red('Could not connect: ' + err));
+  } else {
+    console.log('Client connected');
+
+    // Listen for TestMethod being called from the hub
+    client.onDeviceMethod('TestMethod', function(request, response) {
```

Comment:
The equivalent C# sample doesn't listen for method calls - you can remove this.

Reply:
sure!